### PR TITLE
Unroller error more descriptive when reaching a node without definition

### LINKED ISSUE #5840: Unroller can fail on multi controlled qubits
### Information

- **Qiskit Terra version**: master @ 06dd1552
- **Python version**: 3.7.9
- **Operating system**: macOS Big Sur

### What is the current behavior?

Using the `Unroller` to unroll a circuit can fail, while using `transpile` with `optimization_level=0` works fine.

### Steps to reproduce the problem

```python
from qiskit import transpile
from qiskit.circuit.library import ZGate
from qiskit.transpiler.passes import Unroller

basis = ['u1', 'u2', 'u3', 'cx']
unroller = Unroller(basis)

circuit = ZGate().control().definition  # circuit with CZ gate
unroller(circuit)  # works

circuit = ZGate().control(2).definition  # circuit with CCZ gate
unroller(circuit)  # error, see traceback below
transpile(circuit, basis_gates=basis, optimization_level=0)  # works
```

The error in the unroller is:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/Qiskit/qiskit-terra/qiskit/transpiler/passes/basis/unroller.py in run(self, dag)
     82                 try:
---> 83                     rule = rule[0][0].definition.data
     84                 except (TypeError, AttributeError) as err:

AttributeError: 'NoneType' object has no attribute 'data'

During handling of the above exception, another exception occurred:

QiskitError                               Traceback (most recent call last)
<ipython-input-17-c2529f65b3ed> in <module>
----> 1 Unroller(ubasis)(ZGate().control(2).definition)

~/Qiskit/qiskit-terra/qiskit/transpiler/basepasses.py in __call__(self, circuit, property_set)
    118             self.property_set = property_set_
    119
--> 120         result = self.run(circuit_to_dag(circuit))
    121
    122         result_circuit = circuit

~/Qiskit/qiskit-terra/qiskit/transpiler/passes/basis/unroller.py in run(self, dag)
     83                     rule = rule[0][0].definition.data
     84                 except (TypeError, AttributeError) as err:
---> 85                     raise QiskitError('Error decomposing node {}: {}'.format(node.name, err))
     86
     87             else:

QiskitError: "Error decomposing node p: 'NoneType' object has no attribute 'data'"
```

### What is the expected behavior?

The unroller should be able to unroll any circuit if the basis is sufficient.

### Suggested solutions

Not sure!

### PR DESCRIPTION
**Pull request by TharrmashasthaPV**
### Summary

Fixes #5840 .

### Details and comments

The error raised when the unroller reaches a node without definition is more descriptive now. For instance, for the given example in #5840 , the error message now says "Error decomposing node of instruction 'p': 'NoneType' object has no attribute 'global_phase'. Unable to define instruction 'u' in the given basis."

### CODE REVIEW

File: test/python/transpiler/test_pass_call.py [hunk 1]
```diff
@@ -84,3 +86,16 @@ def test_analysis_pass_remove_property(self):
...
         self.assertIsInstance(property_set, dict)
         self.assertEqual(circuit, result)
+
+    def test_error_unknown_defn_unroller_pass(self):
+        """Check for proper error message when unroller cannot find the definition
+        of a gate."""
+        circuit = ZGate().control(2).definition
+        basis = ['u1', 'u2', 'u3', 'cx']
+        unroller = Unroller(basis)
+        with self.assertRaises(Exception) as cm:
```
Comment (1ucian0) [inline test/python/transpiler/test_pass_call.py:96]:
```suggestion
        with self.assertRaises(QiskitError) as cm:
```

File: test/python/transpiler/test_pass_call.py [hunk 2]
```diff
@@ -84,3 +86,16 @@ def test_analysis_pass_remove_property(self):
...
+        of a gate."""
+        circuit = ZGate().control(2).definition
+        basis = ['u1', 'u2', 'u3', 'cx']
+        unroller = Unroller(basis)
+        with self.assertRaises(QiskitError) as cm:
+            unroller(circuit)
+        exp_msg = repr('Error decomposing node of instruction \'p\': \'NoneType\' object has no' +
+                       ' attribute \'global_phase\'. Unable to define instruction \'u\' in the' +
+                       ' given basis.')
+        self.assertEqual(exp_msg, str(cm.exception))
```
Comment (1ucian0) [inline test/python/transpiler/test_pass_call.py:101]:
```suggestion
        self.assertEqual(exp_msg, cm.exception.message)
```

Comment (TharrmashasthaPV) [reply test/python/transpiler/test_pass_call.py]: The required change has been made.

**Review by 1ucian0 — APPROVED** [review]
Thanks Tharrma!

**Pull request merged**